connexion/lifecycle.py

```diff
@@ -34,4 +34,21 @@ def __init__(self,
         self.mimetype = mimetype
         self.content_type = content_type
         self.body = body
-        self.headers = headers or {}
+        self.headers = ConnexionHeaders(headers)
+
+
+class ConnexionHeaders(object):
```

Comment:
For backward compatibility it should be possible to use the headers object as a map/dict.